src/lib/modes/aead/ascon_aead128/ascon_aead128.h

```diff
@@ -0,0 +1,99 @@
+/*
+* Ascon-AEAD128 AEAD
+* (C) 2025 Jack Lloyd
+*     2025 René Meusel
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_ASCON_AEAD128_H_
+#define BOTAN_ASCON_AEAD128_H_
+
+#include <botan/aead.h>
+
+#include <botan/assert.h>
+#include <botan/internal/ascon_perm.h>
+
+#include <optional>
+
+namespace Botan {
+
+class Ascon_AEAD128_Mode : public AEAD_Mode {
+   public:
+      void set_associated_data_n(size_t idx, std::span<const uint8_t> ad) final;
+
+      bool associated_data_requires_key() const final { return false; }
+
+      std::string name() const final { return "Ascon_AEAD128"; }
+
+      size_t update_granularity() const final { return 1; }
+
+      size_t ideal_granularity() const final { return 32; }
```

Comment:
Seems kind of low tbh

Reply:
Hmm fair enough. Both seem low then :) but that calls for a more systematic evaluation